optaplanner-core/src/main/java/org/optaplanner/core/api/score/constraint/ConstraintMatch.java

```diff
@@ -43,8 +46,12 @@ public ConstraintMatch(String constraintPackage, String constraintName,
             List<Object> justificationList, Score score) {
         this.constraintPackage = constraintPackage;
         this.constraintName = constraintName;
-        this.justificationList = justificationList;
+        this.justificationList = new ArrayList<>(justificationList);
         this.score = score;
+        this.hashCode = (((17 * 37)
+                + constraintPackage.hashCode()) * 37
+                + constraintName.hashCode()) * 37
+                + justificationList.hashCode();
```

Comment:
`justificationList` contains results of `accumulate`, which could be `Set`s with varying contents at varying times. Therefore, `justificationList` *changes from the outside* and we must enforce stable `hashCode` - otherwise all hell breaks loose later.

Reply:
-1 until discussed.
This affects the ConstraintMatch DRL usage and Bavet's ConstraintMatch usage. Do we want that?
Or can we isolate this into a flag on ScoreHolder to active CS-D special stuff?

Reply:
Let's discuss. But it's hard to argue that having inconsistent `hashCode()` is anything else but a bug.